### PR TITLE
Mehrtest

### PR DESCRIPTION
**Pull request by YakMan101**
More test functions and bug fix on short term load api calls

### CODE REVIEW

**Comment by YakMan101** [comment]
Also added fix for the EC2 running out of memory.

File: short_term/async_api_call.py [hunk 1]
```diff
@@ -1,9 +1,40 @@
...
+                }
+        except json.JSONDecodeError as e:
+            return {
+                "error": True,
+                "message": "Invalid JSON response",
+                "response": response.text
+            }
+        except Exception as e:
+            return {
+                "error": True,
```
Comment (mehrdadhalali) [inline short_term/async_api_call.py:32]: Good error handling!

File: dashboard/test_longterm_transform.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    return pd.DataFrame(data)
+
+
+@pytest.mark.parametrize("files, dfs, expected_df", [
+    (["file1.csv", "file2.csv", "file3.csv"],
+     {"file1.csv": pd.DataFrame({'plant_id': [1, 1], 'water_count': [3, 2]}),
+      "file2.csv": pd.DataFrame({'plant_id': [2, 2], 'water_count': [5, 6]}),
+      "file3.csv": pd.DataFrame({'plant_id': [3], 'water_count': [4]})},
+     pd.DataFrame({'plant_id': [1, 1, 2, 2, 3], 'water_count': [3, 2, 5, 6, 4]})),
+    (["file1.csv"], {"file1.csv": pd.DataFrame({'plant_id': [1, 1], 'water_count': [3, 2]})},
```
Comment (Leon-JS02) [inline dashboard/test_longterm_transform.py:31]: Nice range of data here

File: dashboard/test_longterm_transform.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import pytest
+
+import pandas as pd
+import altair as alt
+import pytest
+
+from longterm_transform import concat_csvs, get_times_watered, get_max_temp_per_plant, get_min_moisture_per_plant
+
+
+@pytest.fixture
```
Comment (mehrdadhalali) [inline dashboard/test_longterm_transform.py:14]: Could potentially put this in a conftest file?

**Review by mehrdadhalali — APPROVED** [review]

File: short_term/async_api_call.py [hunk 4]
```diff
@@ -1,9 +1,40 @@
...
 
 
 BASE_URL = "https://data-eng-plants-api.herokuapp.com/plants/"
+TIMEOUT = 45
+
+
+def handle_response(response):
+    """Handles the response from the API call."""
+
+    if response is not None:
```
Comment (Leon-JS02) [inline short_term/async_api_call.py:14]: Indenting could be reduced here by inverting the logic:

```
if response is None:
    return {"error": True, "message": "No response received"}
...

```

**Review by Leon-JS02 — APPROVED** [review]

**Pull request merged**